include/phasar/Utils/Logger.h

```diff
@@ -45,45 +43,131 @@ enum SeverityLevel {
   INVALID
 };
 
-std::string toString(const SeverityLevel &Level);
-
-SeverityLevel toSeverityLevel(const std::string &S);
-
-std::ostream &operator<<(std::ostream &OS, const SeverityLevel &Level);
+class Logger final {
+public:
+  /**
+   * Set the filter level.
+   */
+  static void setLoggerFilterLevel(SeverityLevel Level);
+
+  static SeverityLevel getLoggerFilterLevel();
+
+  static bool isLoggingEnabled();
+
+  static void enable();
+
+  static void disable();
+
+  static llvm::raw_ostream &
+  getLogStream(const std::optional<SeverityLevel> &Level,
+               const std::optional<llvm::StringRef> &Category);
+
+  static bool logCategory(const llvm::StringRef &Category,
+                          const std::optional<SeverityLevel> &Level);
```

Comment:
```suggestion
  static bool logCategory(llvm::StringRef Category,
                          std::optional<SeverityLevel> Level);
```

llvm::StringRef should be passed by value